File: src/jsonl.ts
import { isRecord } from "./types.js";

export function parseJsonLine(line: string): unknown {
  const trimmed = line.trim();
  if (trimmed.length === 0) {
    return undefined;
  }
  const parsed: unknown = JSON.parse(trimmed);
  return parsed;
}

export function parseJsonObject(line: string): Record<string, unknown> | undefined {
  const parsed = parseJsonLine(line);
  if (!isRecord(parsed)) {
    return undefined;
  }
  return parsed;
}

File: python/teams_harness/jsonl.py
from __future__ import annotations

import json
from typing import Any

from .types import is_record


def parse_json_line(line: str) -> Any | None:
    trimmed = line.strip()
    if not trimmed:
        return None
    return json.loads(trimmed)


def parse_json_object(line: str) -> dict[str, Any] | None:
    parsed = parse_json_line(line)
    if not is_record(parsed):
        return None
    return parsed
